### PR TITLE
Allow variants to stay in the now supported `WebP` format during processing

### PR DESCRIPTION
**Pull request by jvhaperen**
Allow ActiveStorage Variants to stay in the now supported `WebP` format during processing, instead of being converted to the fallback PNG format. Done by adding `image/webp` to the list of known web image content types for variants.

### Summary

WebP support was added by https://github.com/rails/rails/pull/38682 so that WebP images could now be handled by default. But when using a variant of a WebP image it was converted to the fallback PNG format, because the WEB_IMAGE_CONTENT_TYPES in ActiveStorage::Variant and ActiveStorage::VariantWithRecord didn't include `image/webp`

### Other Information
The current list of WEB_IMAGE_CONTENT_TYPES (image/png image/jpeg image/jpg image/gif) comes from this commit https://github.com/rails/rails/commit/95117a2ce234c381180429b5b8161fdb44843f30 (@georgeclaghorn committed on 18 Dec 2017). But I couldn't see a reason why image/webp would not be suitable there.

### CODE REVIEW

File: activestorage/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,15 @@
+*   Allow variants to stay in the now supported `WebP` format during processing
+
+    Allow ActiveStorage Variants to stay in the now supported `WebP` format
+    during processing, instead of being converted to the fallback PNG format.
+    Done by adding `image/webp` to the list of known web image content types.
+
+    *Jeroen van Haperen*
+
```
Comment (georgeclaghorn) [inline activestorage/CHANGELOG.md:8]: WebP variant support is new, so no need for a changelog entry.

**Review by georgeclaghorn — CHANGES_REQUESTED** [review]

**Comment by georgeclaghorn** [comment]
Safari accounts for >10% of web users and [doesn’t support WebP yet](https://caniuse.com/#feat=webp).

image

I’m inclined to say we shouldn’t consider WebP a “web image” format until a new version of Safari is released that supports it and most Safari users have upgraded. Thoughts?

File: activestorage/CHANGELOG.md (hunk 1 above)
Comment (jvhaperen) [reply activestorage/CHANGELOG.md:8]: Should I add a commit to remove the full changelog addition? (you selected some lines from the changelog addition I did, so double checking)

**Comment by jvhaperen** [comment]
Hi George, thanks for your reply.

I think WebP specifically is a web format but indeed not 100% supported yet. This would however give the web developer the possibility to decide if it is suitable for the target audience of their application. It would also further complement the already merged PR https://github.com/rails/rails/pull/38682 where WebP handling support was added to Rails.

**Comment by georgeclaghorn** [comment]
It’s broad browser support that makes the existing list of formats safe to pass through unconverted. Until Safari supports WebP, it doesn’t make sense to add to the list—and, for example, render broken WebP image variants by default on all iOS devices.

I’d accept a PR to make the list of web image content types configurable instead, so developers can add WebP if it makes sense for their apps.

**Comment by jvhaperen** [comment]
Ok! I'll go take a look at that.

**Comment by jvhaperen** [comment]
For reference: done in https://github.com/rails/rails/pull/38988

**Comment by georgeclaghorn** [comment]
Closing in favor of #38988.

**Pull request closed without merging**